samples/event_manager_proxy/src/main.c

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <device.h>
+
+#include <ipc/ipc_service.h>
+
+#include <event_manager.h>
+#include <event_manager_proxy.h>
+#include <config_event.h>
+#include <control_event.h>
+#include <measurement_event.h>
+#include <logging/log.h>
+
+#define MODULE main
+
+LOG_MODULE_REGISTER(MODULE);
+
+#define INIT_VALUE1 3
+
+void main(void)
+{
+	int ret;
+	const struct device *ipc_instance  = DEVICE_DT_GET(DT_NODELABEL(ipc0));
+
+	printk("Event Manager Proxy demo started\n");
+	LOG_INF("Starting");
+
+	ret = event_manager_init();
+	if (ret) {
+		LOG_ERR("Event Manager not initialized, err: %d", ret);
+		__ASSERT_NO_MSG(false);
+		return;
+	}
+	LOG_INF("Event manager initialized");
+
+	ret = event_manager_proxy_add_remote(ipc_instance);
+	if (ret) {
+		LOG_ERR("Cannot add remote: %d", ret);
+		__ASSERT_NO_MSG(false);
+		return;
+	}
+	LOG_INF("Event proxy remote added");
+
+	/* Listen to the events from the remote */
+	ret = EVENT_MANAGER_PROXY_REGISTER_LISTENER(ipc_instance, control_event);
```

Comment:
Can we have the remote event tracking registration done in build time?

If we keep it like it is proposed we would need some code (in main or some other central point), at which we would need to execute these registration calls. If we have the registration done at build time we could add a switch to event configuration (kconfig) that would require event tracking from other cores.
If we go this way, there are some open questions - first how to select instance. Maybe we can have short brainstorm on possibilities and tradeoffs?

Reply:
Well - we discussed it offline already. Having it at build time was my idea from the beginning. Current implementation is something that was set by you during the implementation discussion: I agreed to start simple - register it by the name during initialization and then improve it. The instance comes directly from DT. Current implementation is extremely elastic - we can even communicate with other core using UART. The whole logic is here. But I agree that this should be driven towards build time configuration.

Reply:
Actually I meant that we collect information what should be registered at build time while registration is done run time (the difficulty is passing this information between FW binaries).
Maybe we could use dts to create these arrays...

Anyway, we can try to look for possibilities later on.